nitro-protocol/test/contracts/LedgerFinancingApp/requireStateSupported.test.ts

```diff
@@ -0,0 +1,291 @@
+  ],
+} as ParamType;
+
+function appDataABIEncode(appData: LedgerFinancingAppData): string {
+  return ethers.utils.defaultAbiCoder.encode(
+    // ['uint128', 'uint128', 'uint256', 'tuple(address[], uint256[])', 'tuple(address[], uint256[])'],
+    [ledgerFinancingAppDataTy],
+    [appData]
+  );
+}
+
+const initialOutcome = computeOutcome({
+  [MAGIC_NATIVE_ASSET_ADDRESS]: {[intermediary]: 500, [merchant]: 500},
+});
+
+const baseAppData: LedgerFinancingAppData = {
+  // 1/100 -> 1% daily percentage yield.
+  dpyNum: 1,
+  dpyDen: 100,
+  blocknumber: 1,
+  principal: {
+    asset: [MAGIC_NATIVE_ASSET_ADDRESS],
+    amount: [500],
+  },
+  collectedInterest: {
+    asset: [MAGIC_NATIVE_ASSET_ADDRESS],
+    amount: [0],
+  },
+};
+
+const baseState: State = {
+  turnNum: 0,
+  isFinal: false, // intermediary wants to force finalization
+  channelNonce: '0x8',
+  participants,
+  challengeDuration,
+  outcome: initialOutcome,
+  appData: appDataABIEncode(baseAppData),
+  appDefinition: APPDEF,
+};
+
+const variablePart = getVariablePart(baseState);
+
+const signedByMerchant: RecoveredVariablePart = {
+  variablePart,
+  signedBy: BigNumber.from(0b10).toHexString(),
+};
+const signedByIntermediary: RecoveredVariablePart = {
+  variablePart,
+  signedBy: BigNumber.from(0b01).toHexString(),
+};
+const signedByBoth: RecoveredVariablePart = {
+  variablePart,
+  signedBy: BigNumber.from(0b11).toHexString(),
+};
+
+const fixedPart = getFixedPart(baseState);
+
+beforeAll(async () => {
+  ledgerFinancingApp = setupContract(provider, LedgerFinancingAppArtifact, APPDEF);
+});
+
+describe('requireStateSupported', () => {
+  it('accepts unanimous states', async () => {
+    await ledgerFinancingApp.requireStateSupported(fixedPart, [], signedByBoth);
+  });
+
+  it('accepts legitimate interest calculations', async () => {
+    // construct a proof+candidate test case with fair interest calculation, assert passing    // with the interest rate applied.
+    // test case:
+    // - appdata interest rate is 1% per day
+    // - initial outcome is 500:500, with 500 principal
+    // - 1 day passes
+    // - challenge outcome is 505:495
+    advanceOneDay();
+
+    const challengeState: State = {
+      ...baseState,
+      turnNum: baseState.turnNum + 1,
+      outcome: computeOutcome({
+        [MAGIC_NATIVE_ASSET_ADDRESS]: {[intermediary]: 505, [merchant]: 495}, // intermediary picks up 1% of the principal
+      }),
+    };
+    const challengeWithIntermediarySignature: RecoveredVariablePart = {
+      variablePart: getVariablePart(challengeState),
+      signedBy: BigNumber.from(0b01).toHexString(),
+    };
+
+    await ledgerFinancingApp.requireStateSupported(
+      fixedPart,
+      [signedByBoth],
+      challengeWithIntermediarySignature
+    );
+  });
+
+  it('reverts if the proof block number is in the future', async () => {
+    // create a challenge where the intermediary fakes the blocknumber
+    // (in order to get a higher interest rate).
```

Comment:
Wouldn't it be in the interest of the intermediary to fake a block number in the _past_?

Reply:
76f8f179

This test case made sense when `computeOutstandingInterest` was reading the intermediary supplied block number (which was replaced with reading the block number from the chain itself).